RELEASES.md

```diff
@@ -1,3 +1,440 @@
+# Version 3.0 (2020-10-09)
+
+Be prepared for the ink! 3.0 release notes because the whole version was basically a rewrite of
+all the major components that make up ink!. With our experience gained by the last released versions
+of ink! we were able to detect weak spots of the design and provided ink! with more tools,
+more features and more efficiency as ever. Read more below ...
+
+## Just. Be. Rust. 3.0
+
+In the 3.0 update we further explored the space for ink! to just feel like it was plain Rust.
+With this regard we changed the syntax slightly in order to better map from ink! to the generated
+Rust code. So what users see is mostly what will be generated by ink! later.
+
+In this vein `#[ink(storage)]` and `#[ink(event)]` structs as well as `#[ink(message)]` and
+`#[ink(constructor)]` methods now need to be specified with public visibility (`pub`).
+
+The `#[ink(constructors)]` syntax also changes and no longer uses a `&mut self` receiver but
+is now follows the natural Rust constructors scheme. So it is no longer possible to shoot
+yourself in the foot by accidentally forgetting to initialize some important data structures.
+
+**Old ink! 2.0:**
+```rust
+#[ink(storage)]
+fn new_erc20(&mut self, initial_supply: Balance) {
+    let caller = self.env().caller();
+    self.total_supply.set(initial_supply);
+    self.balances.insert(caller, initial_supply);
+}
+```
+**New ink! 3.0:**
+```rust
+#[ink(storage)]
+pub fn new_erc20(initial_supply: Balance) -> Self {
+    let caller = self.env().caller();
+    let mut balances = ink_storage::HashMap::new();
+    balances.insert(caller, initial_supply);
+    Self {
+        total_supply: initial_supply,
+        balances,
+    }
+}
+```
+
+Also ink! 3.0 no longer requires a mandatory `version` field in the header of the ink! module attribute.
+
+Syntactically this is all it takes to port your current ink! smart contracts over to ink! 3.0 syntax.
+
+## New Storage Module
+
+The storage module has been reworked entirely.
+Also it no longer lives in the `ink_core` crate but instead is defined as its own `ink_storage` crate.
+
+In a sense it acts as the standard storage library for ink! smart contracts in that it provides all the
+necessary tools and data structures to organize and operate the contract's storage intuitively and efficiently.
+
+### Lazy
+
+The most fundamental change in how you should think about data structures provided by the new `ink_storage`
+crate is that they are inherently lazy.
```

Comment:
I can relate

Reply:
Would be good to define what we mean by `lazy` near the top here in case it is not immediately obvious. 

(I see that you do a bit further down :point_down:)

Reply:
Hmmm, I think it might be enough to state that we will explain what this means below.

Reply:
done